Tasks/AzureCLIV2/azureclitask.ts

```diff
@@ -54,10 +54,13 @@ export class azureclitask {
             }
 
 
-            if (failOnStdErr) {
+            if (failOnStdErr && aggregatedErrorLines.length > 0) {
+                let error = "";
                 aggregatedErrorLines.forEach((err: string) => {
                     tl.error(err);
+                    error += err;
                 });
+                throw error;
```

Comment:
@issacnitin the sample script which has been mentioned in the issue #12874 will cause exit code to be non zero. And in such case, adding this line is not required. As down below in `finally` block we do check for exit code and fail the task if it is non zero (L86).
Hence I do not completely get this fix. I do understand that we were not throwing in case of failOnStdErr. So should we throw explicitly or does tl.error() will cause task to fail?

Reply:
Yes, so what it means is that the exit code is 0 sometimes, even when the script fails (maybe tool exec issue, as we're setting ignoreReturnCode). But since the customer has set failOnstderr, we need to throw error if any and don't rely on exit code

Reply:
Why tool will return exit code randomly? I doubt it, it should be a consistent behavior for same script.

Reply:
@anuragc617 According to customer's statement the behavior is random

Reply:
IMO `ignoreReturnCode` should not cause this issue. If it is true, the task lib itself will throw an error. We keep is false and handle the exit code ourselves.
Also I don't think it is a tool exec issue. Way too many tasks use tool.exec for it to misbehave without it becoming a big issue. 
I assume we have a very simple one line script repro as mentioned in the issue. can you repro it and locally add more logs to see what is the return code. Ideally it should be non-zero and hence this fix will not fix the issue.

Reply:
Then what could be the issue in your opinion?

Reply:
Not sure. If you have a repro, an you paste debug logs here

Reply:
I am not able to reproduce. let's go ahead with the current hypothesis, and ask customer for debug logs once it happens again.